src/Dev/FunctionalTest.php

```diff
@@ -401,15 +405,28 @@ public function assertExactHTMLMatchBySelector($selector, $expectedMatches, $mes
      */
     public function logInAs($member)
     {
-        if (is_object($member)) {
-            $memberID = $member->ID;
-        } elseif (is_numeric($member)) {
-            $memberID = $member;
-        } else {
-            $memberID = $this->idFromFixture('SilverStripe\\Security\\Member', $member);
+        if (is_numeric($member)) {
+            $member = DataObject::get_by_id(Member::class, $member);
+        } elseif (!is_object($member)) {
+            $member = $this->objFromFixture('SilverStripe\\Security\\Member', $member);
         }
 
-        $this->session()->inst_set('loggedInAs', $memberID);
+        $this->logIn($member);
```

Comment:
Should we make this call logIn() on the current IdentityStore, rather than assuming session?
We should see if we can remove the session assignment and see if we can just rely on Security::setCurrentUser().